src/main/java/bio/terra/service/dataset/flight/datadelete/DatasetDataDeleteFlight.java

```diff
@@ -76,7 +97,15 @@ public DatasetDataDeleteFlight(FlightMap inputParameters, Object applicationCont
     addStep(new CreateExternalTablesStep(bigQueryPdao, datasetService));
 
     // insert into soft delete table
-    addStep(new DataDeletionStep(bigQueryPdao, datasetService, configService));
+    addStep(new DataDeletionStep(bigQueryPdao, datasetService, configService, userReq, autocommit));
+
+    if (!autocommit) {
+      addStep(
+          new TransactionUnlockStep(
+              datasetService, bigQueryPdao, request.getTransactionId(), userReq));
```

Comment:
And then again here - if you're only setting autocommit = false on the case that request.getTransactionId() == null

Reply:
It's the other way around.  I set autocommit to true when request.getTransactionId() is null (e.g. when the user doesn't supply a transaction, it's autocommit